Update CQL and ELM in sample pathways

danlee1025 opened the pull request:
Updated CQL to return boolean values and regenerated ELM for sample pathways.

blangley28 commented:
Few comments:
1. There are errors in the elm (`library.annotation`)
2. This is my oversight from reworking the her2 pathway but the `EndocrineTherapy` node should have some cql. I think this should do it:
```
exists([Procedure: \"Hormone Therapy (procedure) code\"]) or exists([ServiceRequest: \"Hormone Therapy (procedure) code\"])
```
I forget which library these are pulling from but add this to the library:
```
code "Hormone therapy (procedure) code": '169413002' from "SNOMEDCT" display 'Hormone therapy (procedure)'
```
3. This was mentioned in the ticket but not explicitly stated in the DoD but the cql in the pathway should just be the statement name. I think the best way to do this is to take the CQL for the test pathways and create a library then add this as criteria

blangley28 commented:
How does this work with Dylan's PR? Over there the cql in the pathway is just the statement name and then the actual cql comes from a library. I don't think the builder has support for inline cql anymore. I think the easiest solution is to make a cql library for each of the pathways and upload this as criteria so the builder has the necessary criteria

danlee1025 commented:
@blangley28 Ok that makes sense, what about the cql for the preconditions? Should those be included in the cql library also and just include the statement in the pathway?

blangley28 commented:
Since preconditions don't actually do anything yet it doesn't matter although I imagine eventually it will work the same. I'm fine with ignoring those for now

danlee1025 commented:
I added CQL libraries for each of our sample pathways and replaced the inline CQL for the criteria with just the statement name. I left the inline CQL for the preconditions and action nodes since they're not being used right now. The elm in the pathways has also been removed.

dehall reviewed (commented):
So this really highlights how the sample pathways are totally different from the pathways that a user can actually construct. 
For example if you load up one of the sample pathways and look at the branches the criteria isn't populated:
<img width="1569" alt="image" src="https://user-images.githubusercontent.com/13512036/93618338-1d9d7580-f9a5-11ea-847d-e750551888ea.png">


And I don't think this task is the right place to fix all that. My suggestion is that we finish this up by grouping the CQL into libraries by common elements, like we have one already for Tumor Markers, so create a new one for TNM, etc, and then we don't need "pathway-specific" libraries. (The `library` field in the pathway JSON isn't really used anyway, so we can leave that empty for now and at some point maybe remove the field entirely) Then in the pathway definition, update the CQL so it's qualified with the library name. So for example 
```
           "condition": {
             "description": "ERPositive",
             "cql": "ERPositive"
           }
```
would become
```
           "condition": {
             "description": "ERPositive",
             "cql": "Tumor_Markers.ERPositive"
           }
```

Finally in the upcoming task PATHWAY-277 "Implement Import Pathway" we can figure out how to import/export pathways and the linked criteria to make sure everything works.

danlee1025 commented:
Pushed up `TNM.cql` and `Pathologic-Response.cql` and removed pathway specific CQL libraries.

blangley28 commented on public/static/cql/TNM.cql:
```diff
@@ -0,0 +1,81 @@
+        or FHIRHelpers.ToConcept(NLarge.value as FHIR.CodeableConcept) ~ "N3 category (finding)"
+  )
+
+define "T > 2cm":
```
Yea renaming may be helpful - the reason we use cm here is because that's what the pdf pathway has as the transitions

blangley28 reviewed (approved):
LGTM - @dehall is there a task in the backlog for next steps?

dehall commented:
> LGTM - @dehall is there a task in the backlog for next steps?

I think [PATHWAYS-277 - Implement Import Pathway](https://jira.mitre.org/browse/PATHWAYS-277) covers everything under "Must be able to export and then import a pathway without losing/missing anything. " I can add a comment to clarify that the sample pathways should be updated to make sure everything in them works, in particular their included criteria.
